pytorch_lightning/trainer/connectors/training_trick_connector.py

```diff
@@ -28,10 +28,21 @@ def on_trainer_init(
         gradient_clip_algorithm: str,
         track_grad_norm: Union[int, float, str],
         accumulate_grad_batches: Union[int, Dict[int, int]],
-        terminate_on_nan: bool,
+        terminate_on_nan: Optional[bool],
+        detect_anomaly: bool,
     ):
 
-        self.trainer.terminate_on_nan = terminate_on_nan
+        if terminate_on_nan is None:
+            self.trainer.terminate_on_nan = detect_anomaly
```

Comment:
this is confusing changing arg but keeping the internal attributed as it is?

Reply:
This was recommended by @ananthsub: https://github.com/PyTorchLightning/pytorch-lightning/pull/9175#discussion_r697900706
It could be replaced by:
```
if terminate_on_nan is not None:
    rank_zero_deprecation(
        "Trainer argument `terminate_on_nan` was deprecated in v1.5 release"
        " and will be removed in the v1.7 release."
        " Please use trainer argument `detect_anomaly` instead."
    )
self.trainer.terminate_on_nan = terminate_on_nan```

Reply:
I think it makes sense to keep the `terminate_on_nan` attribute internally as we would remove all of its uses when the deprecated argument is removed.